File: video-series-mono/src/main/java/org/jesperancinha/video/repository/VsaRepository.java
package org.jesperancinha.video.repository;

import java.util.HashMap;
import java.util.List;

import lombok.extern.slf4j.Slf4j;
import org.jesperancinha.video.domain.VideoSeries;
import org.springframework.stereotype.Repository;

@Repository
@Slf4j
public class VsaRepository {
    private static final HashMap<String, VideoSeries> videoSeriesStore = new HashMap<>();

    public void saveVideoSeriesHistory(VideoSeries videoSeries) {
        videoSeriesStore.put(videoSeries.getId().toString(), videoSeries);
        log.info("Video series added to hashmap! {}", videoSeries);
        log.info("Recorded history is:");
        videoSeriesStore.forEach((id, recordedVideoSeries) -> log.info("{} - {}", id, recordedVideoSeries));
    }

    public List<VideoSeries> readAll() {
        return videoSeriesStore.values().stream().toList();
    }
}

File: video-series-mono/src/main/kotlin/org/jesperancinha/video/repository/VsaRepository.kt
package org.jesperancinha.video.repository

import org.jesperancinha.video.domain.VideoSeries
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Repository

@Repository
class VsaRepository {
    fun saveVideoSeriesHistory(videoSeries: VideoSeries) {
        videoSeriesStore[videoSeries.id.toString()] = videoSeries
        logger.info("Video series added to hashmap! {}", videoSeries)
        logger.info("Recorded history is:")
        videoSeriesStore.forEach { (id: String?, recordedVideoSeries: VideoSeries?) ->
            logger.info(
                "{} - {}",
                id,
                recordedVideoSeries
            )
        }
    }

    fun readAll(): List<VideoSeries> {
        return videoSeriesStore.values.stream().toList()
    }

    companion object {
        private val videoSeriesStore = HashMap<String, VideoSeries>()
        val logger: Logger = LoggerFactory.getLogger(VsaRepository::class.java)
    }
}